JENKINS-22482: Show total build time

zihaoyu opened the pull request:
by calculating all possible routes in a pipeline, and pick the route with largest total duration value. Total duration value is the sum of durations of each task in that route.

patbos commented:
Hi, Thanks for the PR. Have been busy fixing the 0.9.0 release. Will look at the PR in detail as soon as possible.

patbos commented on src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java:
```diff
@@ -114,7 +119,61 @@ public void setChanges(List<Change> changes) {
+        return totalBuildTime;
+    }
 
+    public void setTotalBuildTime() {
```
Please rename this since it is not a setter method. Suggestion is calculateTotalBuildTime()

zihaoyu replied:
Thanks. Will do.

patbos commented on src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java:
```diff
@@ -114,7 +119,61 @@ public void setChanges(List<Change> changes) {
+        return currentRoute;
+    }
+
+    public void calculatePipelineRoutes(Task task, final Route route, List<Route> allRoutes) {
```
Should this method really be public?

zihaoyu replied:
I wanted to test this method directly so I made it public. Do you suggest making it private?

marcusphi replied:
Why not default, aka package-private?

Marcus Philip
Twitter: @marcus_phi

> 8 jun 2015 kl. 02:35 skrev Zihao Yu <notifications@github.com>:
> 
> In src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java:
> 
> > +                }
> > +            }
> > +            this.totalBuildTime = maxTime;
> > +        }
> > +    }
> > +
> > +    private Route createRouteAndCopyTasks(final Route route, Task task) {
> > +        Route currentRoute = new Route();
> > +        if (route != null) {
> > +            currentRoute.setTasks(newArrayList(route.getTasks()));
> > +        }
> > +        currentRoute.addTask(task);
> > +        return currentRoute;
> > +    }
> > +
> > +    public void calculatePipelineRoutes(Task task, final Route route, List<Route> allRoutes) {
> I wanted to test this method directly so I made it public. Do you suggest making it private?
> 
> —
> Reply to this email directly or view it on GitHub.
>

patbos commented:
Have looked through the PR. Great job.
Please squash the commits into one and add Jira issue number in the commit message.
Please add src/main/resources/se/diabol/jenkins/pipeline/DeliveryPipelineView/help-showTotalBuildTime.html with a help text describing the feature. This will show up in the configuration page of the view.

zihaoyu commented:
@patbos Thanks for the feedback. I have addressed the comments:
- rename setter method
- change visibility of the method
- add help message
- squash commits into one and add JIRA number

Please take another look. Thanks!

patbos commented:
:+1:
